proposals/p2287.md

```diff
@@ -0,0 +1,214 @@
+# Allow unqualified name lookup
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/####)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+    -   [Classes](#classes)
+    -   [Interfaces](#interfaces)
+    -   [Namespaces](#namespaces)
+    -   [impls](#impls)
+-   [Open question](#open-question)
+    -   [Implicit instance binding to `me`](#implicit-instance-binding-to-me)
+    -   [Out-of-line definitions for impls](#out-of-line-definitions-for-impls)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [No unqualified lookup when defining outside a scope](#no-unqualified-lookup-when-defining-outside-a-scope)
+
+<!-- tocstop -->
+
+## Abstract
+
+Allow unqualified name lookup in multiple situations:
+
+-   For classes and interfaces, whether inside the class scope or within an
+    out-of-line function definition.
+-   For namespaces, when the namespace is used in a declaration.
+
+## Problem
+
+[Member access](/docs/design/expressions/member_access.md) defines certain
+member access behaviors. However, it doesn't cover what happens if an
+unqualified name lookup occurs within a class, particularly for an out-of-line
+member function definition, or other situations.
+
+## Background
+
+The [member access design](/docs/design/expressions/member_access.md) and
+[information accumulation principle](/docs/project/principles/information_accumulation.md)
+affect this.
+
+This will also work similarly to
+[unqualified lookup within C++](https://en.cppreference.com/w/cpp/language/unqualified_lookup).
+
+## Proposal
+
+Allow unqualified name lookup which will use the appropriate scope.
```

Comment:
It feels a little funny to have the abstract be more concrete than the proposal section. Maybe swap the content of the two sections?